yt_dlp/extractor/tver.py

```diff
@@ -171,7 +166,87 @@ def _real_extract(self, url):
             'description': str_or_none(video_info.get('description')),
             'thumbnails': thumbnails,
+        }
+
+        backend = self._configuration_arg('backend', ['streaks'])[0]
+
+        if backend not in ('brightcove', 'streaks'):
+            raise ExtractorError(f'Invalid backend value: {backend}', expected=True)
+
+        if backend == 'brightcove':
+            return self._brightcove_backend(data, video_info)
+
+        return self._streaks_backend(data, video_info, video_id)
+
+    def _brightcove_backend(self, result, video_info):
+        self.write_debug('Using Brightcove backend')
+
+        p_id = video_info['video']['accountID']
+        r_id = traverse_obj(video_info, ('video', ('videoRefID', 'videoID')), get_all=False)
+
+        if not r_id:
+            raise ExtractorError('Failed to extract reference ID for Brightcove')
+
+        if not r_id.isdigit():
+            r_id = f'ref:{r_id}'
+
+        result.update({
+            '_type': 'url_transparent',
             'url': smuggle_url(
                 self.BRIGHTCOVE_URL_TEMPLATE % (p_id, r_id), {'geo_countries': ['JP']}),
             'ie_key': 'BrightcoveNew',
-        }
+        })
+
+        return result
+
+    def _streaks_backend(self, result, video_info, video_id):
+        self.write_debug('Using streaks.jp backend')
+
+        ref_id = traverse_obj(video_info, ('streaks', 'videoRefID'), get_all=False)
+        project_id = traverse_obj(video_info, ('streaks', 'projectID'), get_all=False)
+
+        if not ref_id:
+            raise ExtractorError('Failed to extract reference ID for streaks.jp stream info')
+
+        if not project_id:
+            raise ExtractorError('Failed to extract project ID for streaks.jp stream info')
+
+        if not ref_id.isdigit() and not ref_id.startswith('ref:'):
+            ref_id = f'ref:{ref_id}'
+
+        url = self.STREAKS_URL_TEMPLATE % (project_id, ref_id, 'aa')
+        self.write_debug(f'Streaks URL: {url}')
+
+        json_info = self._download_json(
+            url,
+            video_id,
+            'Downloading streaks.jp streams video info',
+            headers={
+                'origin': 'https://tver.jp/',
+                'referer': 'https://tver.jp/',
+                **self.geo_verification_headers(),
+            },
+        )
+
+        res = traverse_obj(json_info, ('sources', 0, 'resolution'), default=None)
+        m3u8_url = traverse_obj(json_info, ('sources', 0, 'src'), default=False)
+        if not m3u8_url:
+            raise ExtractorError('Failed to extract m3u8 URL')
+
+        formats, subtitles = self._extract_m3u8_formats_and_subtitles(
+            m3u8_url,
+            video_id,
+            'mp4',
+            m3u8_id='hls',
+            quality=res,
+            headers={'origin': 'https://tver.jp/', 'referer': 'https://tver.jp/'},
+            note='Downloading streaks.jp m3u8 information',
+        )
+
+        result.update({
+            'id': video_id,
```

Comment:
It looks like episode id is used instead of videoRefID.

Reply:
I discussed this behavior with @bashonly in discord we agreed as the streaks id is unique per site, using them as unique id is meaningless. And using tver id as unique identifier makes more sense now.

Reply:
I tested that only TX (tv-tokyo)'s video id is different in brightcove and streaks . other tv station's video id is the consistent between brightcove and streaks. 

I'm not sure if there's a situation that two episodeId have a same videoRefID (for example: the episode is replayed after one or two years. or they will reuse the original episode id? )

Reply:
IMO it didn't make sense to use brightcove's internal `ref:xxxx` `videoRefID` instead of episode_id as video ID for TVer, either. I wish we can change both to episode ID.

Reply:
It doesn't make sense to use release_date as when the content was uploaded to the service, thats what the timestamp/upload_date is for. 

When i or anyone uses release date i assume the mean when said content was made not when it was upload to provider service. @bashonly could give guidance on this. 

About blindly the year. i haven't encountered any tver link yet where the d-m was given but the content was released in earlier year. If you have example please list them. The code you see i have been using it for a year so, and i follow mostly variety shows. they do release old episodes but if it's not broadcasted in the same year they ONLY list the year.